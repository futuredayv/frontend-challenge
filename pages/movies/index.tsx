//#region Global Imports
import * as React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';
//#endregion Global Imports

//#region Local Imports
import './style.scss';
//#endregion Local Imports

//#region Interface Imports
import { IMovies } from '@Interfaces';
//#endregion Interface Imports

class Movies extends React.Component<IMovies.IProps, IMovies.IState> {

	constructor(props: IMovies.IProps) {
		super(props);

		this.state = {
		};
	}

	public render(): JSX.Element {
		return (
			<div className="Movies">
				Movies
			</div>
		);
	}
}

const mapStateToProps = state => state;

const mapDispatchToProps = (dispatch: Dispatch) => (
);

export default connect(mapStateToProps, mapDispatchToProps)(Movies);

