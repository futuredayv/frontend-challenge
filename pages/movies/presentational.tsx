import * as React from 'react';
import { IMovies } from '@Interfaces';

import { Layout, Search, Toast } from '@Components';

export class Presentational extends React.Component<IMovies.IProps> {
	render() {
		const { err, isLoading, children } = this.props;

		return (
			<Layout title="Popular Movies">
				{isLoading || !!err ? (
					<Toast isLoading={isLoading} err={err} />
				) : (
					<>
						<Search />
						<div className="grid-area">{children}</div>
					</>
				)}
			</Layout>
		);
	}
}
