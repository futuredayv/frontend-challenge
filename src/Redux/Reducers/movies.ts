//#region Local Imports
import { ActionConsts } from '@Definitions';
//#endregion Local Imports

//#region Interface Imports
import { IAction, IMovies } from '@Interfaces';
//#endregion Interface Imports

/**
 * INITIAL_STATE
 */
const INITIAL_STATE: IMovies.IStateProps = {};

type IMapPayload = IMovies.Actions.IMapPayload;

export const MoviesReducer = (
	state = INITIAL_STATE,
	action: IAction<IMapPayload>,
) => {
	switch (action.type) {
		case ActionConsts.Movies.SetReducer:
			return {
				...state,
				...action.payload,
			};

		case ActionConsts.Movies.ResetReducer:
			return INITIAL_STATE;

		case ActionConsts.Movies.FetchJSON:
			return {
				...state,
				isLoading: true,
			};

		case ActionConsts.Movies.FetchJSON_SUCCESS:
			const { payload: data } = action;

			return {
				...state,
				isLoading: false,
				data,
			};

		case ActionConsts.Movies.FetchJSON_FAIL:
			const { err } = action.payload;

			return {
				...state,
				isLoading: false,
				err,
			};

		default:
			return state;
	}
};
