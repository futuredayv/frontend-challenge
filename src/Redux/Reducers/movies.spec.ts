import { ActionConsts } from '@Definitions';

import { MoviesReducer, INITIAL_STATE } from './movies';

describe('Reducers', () => {
	describe('Movies Reducer', () => {
		it('should handle FetchJSON', () => {
			expect(
				MoviesReducer(INITIAL_STATE, {
					type: ActionConsts.Movies.FetchJSON,
				}),
			).toMatchSnapshot()
		});

		it('should handle FetchJSON_SUCCESS', () => {
			expect(
				MoviesReducer(INITIAL_STATE, {
					type: ActionConsts.Movies.FetchJSON_SUCCESS,
					payload: {
						movies: [{ title: 'John Wick' }],
						series: [{ title: 'Sherlock' }],
					}
				}),
			).toMatchSnapshot()
		});

		it('should handle Fetch_JSON_FAIL', () => {
			expect(
				MoviesReducer(INITIAL_STATE, {
					type: ActionConsts.Movies.FetchJSON_FAIL,
					payload: {
						isLoading: false,
						err: 'Oops! Something Went Wrong.',
					}
				}),
			).toMatchSnapshot()
		});

		it('should handle UpdateFilterOptions', () => {
			expect(
				MoviesReducer(INITIAL_STATE, {
					type: ActionConsts.Search.UpdateFilterOptions,
					payload: {
						search: 'john',
						sort: {
							by: 'title',
							ordering: 'ASC',
						},
					},
				}),
			).toMatchSnapshot()
		});
	});
});
