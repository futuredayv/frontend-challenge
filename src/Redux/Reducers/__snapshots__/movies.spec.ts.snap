// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducers Movies Reducer should handle Fetch_JSON_FAIL 1`] = `
Object {
  "err": "Oops! Something Went Wrong.",
  "filterOptions": Object {
    "search": "",
    "sort": Object {
      "by": "title",
      "ordering": "ASC",
    },
  },
  "isLoading": false,
  "movies": Array [],
  "series": Array [],
}
`;

exports[`Reducers Movies Reducer should handle FetchJSON 1`] = `
Object {
  "err": "",
  "filterOptions": Object {
    "search": "",
    "sort": Object {
      "by": "title",
      "ordering": "ASC",
    },
  },
  "isLoading": true,
  "movies": Array [],
  "series": Array [],
}
`;

exports[`Reducers Movies Reducer should handle FetchJSON_SUCCESS 1`] = `
Object {
  "err": "",
  "filterOptions": Object {
    "search": "",
    "sort": Object {
      "by": "title",
      "ordering": "ASC",
    },
  },
  "isLoading": true,
  "movies": Array [
    Object {
      "title": "John Wick",
    },
  ],
  "series": Array [
    Object {
      "title": "Sherlock",
    },
  ],
}
`;

exports[`Reducers Movies Reducer should handle UpdateFilterOptions 1`] = `
Object {
  "err": "",
  "filterOptions": Object {
    "search": "john",
    "sort": Object {
      "by": "title",
      "ordering": "ASC",
    },
  },
  "isLoading": true,
  "movies": Array [],
  "series": Array [],
}
`;
